src/verinfast/agent.py

```diff
@@ -583,16 +580,10 @@ def scanRepos(self):
             repos = self.config.config["repos"]
             if repos:
                 for repo_url in [r for r in repos if len(r) > 0]:       # ignore blank lines from server
-                    match = re.search(r"([^/]*\.git.*)", repo_url)
-                    if match:
-                        repo_name = match.group(1)
-                    else:
-                        repo_name = repo_url.rsplit('/', 1)[-1]
-                    if "@" in repo_name and re.search(r"^.*@.*\..*:", repo_url):
-                        repo_url = "@".join(repo_url.split("@")[0:2])
-                    elif "@" in repo_name:
-                        repo_url = repo_url.split("@")[0]
+                    repo_name, repo_url, branch = get_repo_name_url_and_branch(repo_url)
```

Comment:
it's really bad form to return tuples. Technically allowed, but better to return dicts. that way order doesn't matter and each thing is named.

Reply:
Fixed